crates/sargon/src/profile/v100/profile.rs

```diff
@@ -234,48 +234,167 @@ impl<T: IsEntity> IdentifiedVecOf<T> {
     }
 }
 
+/// Analysis has identified that `entity1` and `entity2` have the same `FactorInstance`
+/// in common. Either a `TransactionSigning` instance or an `AuthenticationSigning` instance.
+/// Either in `EntitySecurityState::Unsecure` or in `EntitySecurityState::Secure`
+///
+/// Where: `entity1.address() != entity2.address()`
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct DuplicateInstances {
+    /// One of the entities containing `factor_instance`
+    ///
+    /// `entity1.address() != entity2.address()`
+    pub(crate) entity1: AccountOrPersona,
+
+    /// The other entity containing `factor_instance`
+    ///
+    /// `entity1.address() != entity2.address()`
+    pub(crate) entity2: AccountOrPersona,
+
+    /// The FactorInstance which is shared between `entity1` and `entity2`
+    pub(crate) factor_instance: FactorInstance,
+}
+impl Identifiable for DuplicateInstances {
+    type ID = FactorInstance;
+    fn id(&self) -> Self::ID {
+        self.factor_instance.clone()
+    }
+}
+
+/// Like `DuplicateInstances` but also contains a flag if the current OS is Android.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct DuplicateInstancesWithKnownOs {
+    /// The duplicated instances found
+    pub(crate) duplicate_instances: DuplicateInstances,
+    /// If HostInfo driver returns that the current OS is Android
+    pub(crate) is_android: bool,
```

Comment:
because should REALLY react to this scenario if we were to detect it on iOS, was my thinking. But I realize this is only relevant for us... so can only happen if we are given an already ""corrupt"" Profile and run it with cable. hmm